### ISSUE TITLE
Suggest adding wasm32-unknown-unknown to cargo config on Linux

### ISSUE THREAD

**Issue by ealmloff** [DioxusLabs/dioxus#1796]
### Discussed in https://github.com/DioxusLabs/dioxus/discussions/947

<div type='discussions-op-text'>

<sup>Originally posted by **pickfire** April  9, 2023</sup>
I noticed if I did not specify `wasm32-unknown-unknown` and I use `dioxus serve --hot-reload` with `rust-analyzer` running in the terminal, it needs a full build every time and the incremental build is very slow.

But if I add the wasm target to cargo config in `.cargo/config`, then it became very fast even on non-rsx changes (from ~48s to ~2.5s). Maybe because there are some issues with caching for rust-analyzer?

```toml
[build]
target = "wasm32-unknown-unknown"
```

Maybe we can suggest adding that to https://dioxuslabs.com/docs/0.3/guide/en/getting_started/hot_reload.html if web is used?</div>

This issue does not effect windows or mac, but there have been other reports of the same issue on discord about hot reloading doing a full rebuild on linux with rust analyzer

**Comment by ealmloff** [comment]
We should change `rust-analyzer.cargo.target`, not the default target for cargo so that we don't break `cargo run` for any cross platform apps

**Comment by ealmloff** [comment]
We could try to insert it into the config in https://github.com/rust-lang/rust-analyzer/issues/13529

**Comment by Craig-Macomber** [comment]
> [build]
> target = "wasm32-unknown-unknown"

I manually added that to my cargo config (it seemed the easiest way at the time I could get syntax highlighting for web specific stuff in vscode), and I noticed it breaks `dx serve --platform desktop`, giving "I/O Error: Exec format error (os error 8)" (I think because it compiled for wasm).

I think thats a bug in `dx serve` : it seems like if you explicitly select desktop, it probably shouldn't build wasm.

On the bright side, with the great changes in 0.5, I no longer have significant wasm specific code and use desktop for debugging, so I'm just going to remove that from my cargo.toml.
